pkg/security/module/module.go

```diff
@@ -274,12 +312,39 @@ func (m *Module) Reload() error {
 	m.displayReport(report)
 
 	// report that a new policy was loaded
-	monitor := m.probe.GetMonitor()
-	monitor.ReportRuleSetLoaded(ruleSet, loadErr)
+	monitor.ReportRuleSetLoaded(ruleSetLoadedReport)
 
 	return nil
 }
 
+// Reload the rule set
+func (m *Module) Reload() error {
+	var selfTestPolicy *rules.Policy
+	if m.selfTester != nil {
+		selfTestPolicy = m.selfTester.GetSelfTestPolicy()
+	}
+
+	return m.reloadWithPoliciesDir(m.config.PoliciesDir, selfTestPolicy)
+}
+
+func (m *Module) doSelfTest() error {
```

Comment:
This should live inside the `RunSelfTest()` method. Afaik, you don't need the module here.